src/libgit2/repository.c

```diff
@@ -3246,14 +3246,18 @@ int git_repository_set_workdir(
 	if (git_fs_path_prettify_dir(&path, workdir, NULL) < 0)
```

Comment:
I have one comment about this call as well. When checking its implementation there are 2 paths where we do not deallocate the buffer if the function fails, while there's one where we correctly deallocate `path`.

Here where we do not deallocate in case of error:
https://github.com/libgit2/libgit2/blob/99018a0f35e7a85001626d7038185cbdec712192/src/util/fs_path.c#L389
https://github.com/libgit2/libgit2/blob/99018a0f35e7a85001626d7038185cbdec712192/src/util/fs_path.c#L403

I apologise if I am wrong (not too familiar with libgit2 codebase), but I would assume a function does clean up in case of error for all paths, and because of this a fix like this:
```
if (git_fs_path_prettify_dir(&path, workdir, NULL) < 0) {
    git_str_dispose(&path);
    return -1;
}
```
won't work because we might hit a double-free problem.